internal/services/pubkey_service.go

```diff
@@ -79,7 +91,7 @@ func CreatePubkey(w http.ResponseWriter, r *http.Request) {
 }
 
 func ListPubkeys(w http.ResponseWriter, r *http.Request) {
-	pubkeyDao, err := dao.GetPubkeyDao(r.Context(), db.DB)
+	pubkeyDao, err := GetPubkeyDao(r.Context())
```

Comment:
It depends on which layer you want to stub things - if you want to replace DAO with stubbed functions, then no change is required on this line. You can simply pass `db.DB` which will be probably `nil` and it's a non-issue as you will not use it at all.

Or you can choose to stub at the driver level, then implement a new `db.Transaction` and pass it into sqlx. But this is much more challenging option, I think you want to replace DAO instead.